### ISSUE TITLE
2.14.0: included relationship in JSONAPISerializer with additional links.related causes backtracking re-render

### ISSUE THREAD

**Issue by buschtoens** [emberjs/data#5023]
Demo app here: https://github.com/buschtoens/ember-data-included-records-demo  
Ember Data 2.13 works fine.

When a JSONAPI compliant server responds with included relationships, but also includes a `links.related` field, Ember crashes with a [backtracking re-render error](https://github.com/emberjs/ember.js/issues/13948), if that relationship was accessed in a template.

```js
{
  "data": {
    "type": "authors",
    "id": "1",
    "attributes": {
      "age": 20,
      "last-name": "Corkery",
      "first-name": "Joseph"
    },
    "relationships": {
      "blog-posts": {
        "data": [
          {
            "type": "blog-posts",
            "id": "1"
          },
          {
            "type": "blog-posts",
            "id": "2"
          }
        ],
        "links": {
          "related": "/show-stopper" // <=== this is the culprit
        }
      }
    }
  },
  "included": [
    {
      "type": "blog-posts",
      "id": "1",
      "attributes": {
        "text": "Enim et nihil. Nihil optio voluptas blanditiis. Officiis dignissimos consequatur eum. Et et eveniet sapiente voluptas soluta optio est.",
        "title": "aut odit voluptas"
      },
      "relationships": {
        "author": {
          "data": {
            "type": "authors",
            "id": "1"
          }
        }
      }
    },
    {
      "type": "blog-posts",
      "id": "2",
      "attributes": {
        "text": "Dolores ab consequatur exercitationem. Alias nisi quod sint animi iusto doloremque blanditiis. Enim accusantium animi et officia.",
        "title": "unde et aut"
      },
      "relationships": {
        "author": {
          "data": {
            "type": "authors",
            "id": "1"
          }
        }
      }
    }
  ]
}
```

A request to `/show-stopper` is never actually performed.

**Comment by buschtoens** [comment]
This was quite hard to track down. :slightly_frowning_face: 

Most likely #5017 is exactly this problem.

**Comment by buschtoens** [comment]
Possibly a duplicate of #4942.

**Comment by tsteuwer** [comment]
We are having the same issue in all of our apps. We've had to go back to 2.13.1

**Comment by Charizard** [comment]
Same here!! Going back to 2.13

**Comment by rtablada** [comment]
I also ran into this issue when the API returns `links` for a relation (included or not in my case).

**Comment by rtablada** [comment]
This issue is happening in ember-data `2.14.0` through `2.14.3`

**Comment by giano** [comment]
Same issue. 2.14.3

**Comment by rstudner** [comment]
Any update on this?  This essentially blocks upgrading to 2.14.x for many apps.

**Comment by ctcpip** [comment]
This issue should be closed.  It's a duplicate of #4942

**Comment by mmun** [comment]
Hi @buschtoens. I tested my fix (#5119) against your branch and the data loads correctly. Thank you for creating this example app!
